radix-common/src/crypto/ed25519/private_key.rs

```diff
@@ -11,37 +11,37 @@ impl Ed25519PrivateKey {
     pub const LENGTH: usize = 32;
 
     pub fn public_key(&self) -> Ed25519PublicKey {
-        Ed25519PublicKey(PublicKey::from(&self.0).to_bytes())
+        Ed25519PublicKey(SigningKey::from(&self.0).verifying_key().to_bytes())
```

Comment:
I wonder if we should probably change to wrap a `SigningKey` here?

It would save us recreating it a lot when we do lots of signing/public key extraction, and `SigningKey` implements `Zeroize`, but `SecretKey` doesn't - which seems to suggest that `SigningKey` is intended to be the in-memory signing key type they intend us to use.

Reply:
Yeah, that makes sense. I will refactor it.

Reply:
Done